File: Cis 400 Website/my-new-app/src/scripts/OddsApiScript.js
import axios from 'axios';

const fetchOddsData = async () => {
  const API_KEY = 'b00dcb637ff2c90f0acf59dd1bf7fbac'; // Replace with your API key
  const SPORT = 'americanfootball_nfl';
  const REGIONS = 'us';
  const MARKETS = 'h2h,spreads,totals';
  const ODDS_FORMAT = 'american';
  const DATE_FORMAT = 'iso';

  try {
    const response = await axios.get(
      `https://api.the-odds-api.com/v4/sports/${SPORT}/odds`,
      {
        params: {
          api_key: API_KEY,
          regions: REGIONS,
          markets: MARKETS,
          oddsFormat: ODDS_FORMAT,
          dateFormat: DATE_FORMAT,
        },
      }
    );

    console.log('Number of events:', response.data.length);
    return response.data; // Return the JSON data
  } catch (error) {
    console.error('Failed to fetch odds:', error.message);
    throw error;
  }
};

export default fetchOddsData;

File: OddsAPIScript.py
import requests
import json


# An api key is emailed to you when you sign up to a plan
# Get a free API key at https://api.the-odds-api.com/
API_KEY = '6448192984900355b8ceaf3388466e6b'

SPORT = 'basketball_nba' # use the sport_key from the /sports endpoint below, or use 'upcoming' to see the next 8 games across all sports

REGIONS = 'us' # uk | us | eu | au. Multiple can be specified if comma delimited

MARKETS = 'h2h,spreads,totals' # h2h | spreads | totals. Multiple can be specified if comma delimited

ODDS_FORMAT = 'american' # decimal | american

DATE_FORMAT = 'iso' # iso | unix


# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # 
#
# Now get a list of live & upcoming games for the sport you want, along with odds for different bookmakers
# This will deduct from the usage quota
# The usage quota cost = [number of markets specified] x [number of regions specified]
# For examples of usage quota costs, see https://the-odds-api.com/liveapi/guides/v4/#usage-quota-costs
#
# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # 

odds_response = requests.get(
    f'https://api.the-odds-api.com/v4/sports/{SPORT}/odds',
    params={
        'api_key': API_KEY,
        'regions': REGIONS,
        'markets': MARKETS,
        'oddsFormat': ODDS_FORMAT,
        'dateFormat': DATE_FORMAT,
    }
)

if odds_response.status_code != 200:
    print(f'Failed to get odds: status_code {odds_response.status_code}, response body {odds_response.text}')

else:
    odds_json = odds_response.json()
    print('Number of events:', len(odds_json))
    print(odds_json)

    # Check the usage quota
    print('Remaining requests', odds_response.headers['x-requests-remaining'])
    print('Used requests', odds_response.headers['x-requests-used'])

    # Specify the filename
    filename = "oddsapi_raw_data.json"

    # Write JSON data to a new file
    with open(filename, "w") as file:
        json.dump(odds_json, file, indent=4)
